### ISSUE TITLE
tests failing after integration of viewzoom/viewrotation code

### ISSUE THREAD

**Issue by achadwick** [mypaint/mypaint#818]
Apologies for the empty initial message! I clicked too fast.

I have tried to fix up the failing tests in `tests.mypaintlib.DocPaint.test_docpaint()` and elsewhere, but even with the following, I cannot get a generated `test_docPaint_flat.png` to resemble `tests/correct_docPaint_flat.png`). I'm aware from #816 that the zoom being passed in is probably wrong. Can you have a look at it please, since this is your new code?

(Diff is against master as of 06282c7c5b46e5aac6d0e098297d3d82148c7774, and represents me trying to fix up the tests for the newly integrated code)

```diff
diff --git a/tests/mypaintlib.py b/tests/mypaintlib.py
index 5ff29d04..cbeae9e7 100755
--- a/tests/mypaintlib.py
+++ b/tests/mypaintlib.py
@@ -93,7 +93,16 @@ class Painting (unittest.TestCase):
                 dtime = t - t_old
                 t_old = t
                 s.begin_atomic()
-                b.stroke_to(s.backend, x * 4, y * 4, pressure, 0.0, 0.0, dtime)
+                b.stroke_to(
+                    s.backend,
+                    x * 4,
+                    y * 4,
+                    pressure,
+                    0.0, 0.0,
+                    dtime,
+                    1.0,  # view zoom
+                    0.0,  # view rotation
+                )
                 s.end_atomic()
         print('%0.4fs, ' % (time() - t0,), end="", file=sys.stderr)
         # FIXME: why is this time so different each run?
@@ -209,7 +218,7 @@ class DocPaint (unittest.TestCase):
     @classmethod
     def tearDownClass(cls):
         os.chdir(cls._old_cwd)
-        shutil.rmtree(cls._temp_dir, ignore_errors=True)
+        # shutil.rmtree(cls._temp_dir, ignore_errors=True)
 
     def test_docpaint(self):
         """Saved and reloaded documents look identical"""
@@ -236,7 +245,15 @@ class DocPaint (unittest.TestCase):
             t_old = t
 
             layer = doc.layer_stack.current
-            layer.stroke_to(doc.brush, x / 4, y / 4, pressure, 0.0, 0.0, dtime)
+            layer.stroke_to(
+                doc.brush,
+                x / 4, y / 4,
+                pressure,
+                0.0, 0.0,
+                dtime,
+                1.0,  # view zoom
+                0.0,  # view rotation
+            )
 
             # Vary the colour so we know roughly where we are
```

EDIT: the `tearDown()` amendment is just me wanting to take a look at how the output differs.

**Comment by achadwick** [comment]
The current state of master at 06282c7c5b46e5aac6d0e098297d3d82148c7774 is as <https://travis-ci.org/mypaint/mypaint/builds/244006885#L1458>. Pretty simply, it's a `stroke_to()` argument count thing which *should* be easily fixable.

```
======================================================================
ERROR: test_docpaint (tests.mypaintlib.DocPaint)
Saved and reloaded documents look identical
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/mypaint/mypaint/tests/mypaintlib.py", line 239, in test_docpaint
    layer.stroke_to(doc.brush, x / 4, y / 4, pressure, 0.0, 0.0, dtime)
TypeError: stroke_to() takes exactly 10 arguments (8 given)
======================================================================
ERROR: test_brush_paint (tests.mypaintlib.Painting)
30s of painting at 4x with a charcoal brush
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/mypaint/mypaint/tests/mypaintlib.py", line 96, in test_brush_paint
    b.stroke_to(s.backend, x * 4, y * 4, pressure, 0.0, 0.0, dtime)
  File "/home/travis/build/mypaint/mypaint/lib/mypaintlib.py", line 285, in stroke_to
    def stroke_to(self, *args): return _mypaintlib.PythonBrush_stroke_to(self, *args)
TypeError: PythonBrush_stroke_to() takes exactly 10 arguments (8 given)
----------------------------------------------------------------------
```

**Comment by achadwick** [comment]
With the changes in the diff, I get the following error from `python setup.py test`:

```
======================================================================
FAIL: test_docpaint (tests.mypaintlib.DocPaint)
Saved and reloaded documents look identical
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/andrewc/Development/MyPaintProject/mypaint/tests/mypaintlib.py", line 352, in test_docpaint
    join(paths.TESTS_DIR, 'correct_docPaint_flat.png'),
  File "/home/andrewc/Development/MyPaintProject/mypaint/tests/mypaintlib.py", line 188, in assert_pngs_equal
    self.assertTrue(equal_enough, msg=msg)
AssertionError: 'test_docPaint_flat.png' and '/home/andrewc/Development/MyPaintProject/mypaint/tests/correct_docPaint_flat.png' have different sizes ((320, 256, 3), (256, 320, 3))

----------------------------------------------------------------------
```

And the output is visibly different:

![test_docpaint_flat](https://user-images.githubusercontent.com/61299/27254456-37a39852-5380-11e7-88d7-d363deda2533.png)
![correct_docpaint_flat](https://user-images.githubusercontent.com/61299/27254457-37a84d84-5380-11e7-93c0-80e5a58c0d3d.png)

----------------------------

Our goal here should be to EITHER find the right values to plug into `stroke_to()` to make the output look the same, OR investigate why the test brushes fail like that (and make new ones), OR to burn our bridges and push a new `correct_*.png` set.

**Comment by briend** [comment]
Ah, crud. Sorry about that.  At 100% zoom and zero rotation all the brushes should be totally identical.  It looks like the test data that you fixed for me has zero for zoom, so it totally cancels out the speed input which that watercolor brush is using.  I'll try to get a patch ready shortly.  Thanks for fixing my mess! ;-)

**Comment by briend** [comment]
Just realized your diff uses 1.0 for zoom.  eek.  Oh...  that's right, I removed a division by brush radius here:
https://github.com/mypaint/libmypaint/blob/master/mypaint-brush.c#L503
So, yeah, I changed the fundamental way that speed is calculated even at 100% zoom.  My theory was the dividing by brush_radius was a workable alternative to having zoom calibration, so doing both certainly doesn't makes sense.

Yes, all brushes that use speed will act a bit differently, but more consistently across zoom levels.  Previously if you zoomed in a bit you would lose your speed (unless you shrank your brush radius).

SO, I think option 3 is the best and just push a new set of images.  I'll try to get that pull ready.
